### PR TITLE
Clean-up around MeshObject

### LINKED ISSUE #21848: Clean-up around MeshObject
Follow on from #12827.

Mainly removing unused code.

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #21848

Most of these changes are around type safety and cleaning up the API. Largely I have not addressed any performance related issues. This addresses issues that are relevant to the use of `MeshObject` in `NexusGeometry`

**Tester**
A code review should be sufficient.

### CODE REVIEW

File: Framework/Geometry/src/Objects/MeshObject.cpp [hunk 1]
```diff
@@ -384,10 +381,10 @@ double MeshObject::solidAngle(const Kernel::V3D &observer,
 {
   std::vector<V3D> scaledVertices;
   scaledVertices.reserve(m_vertices.size());
-  for (size_t i = 0; i < m_vertices.size(); ++i) {
-    scaledVertices.push_back(V3D(scaleFactor.X() * m_vertices[i].X(),
-                                 scaleFactor.Y() * m_vertices[i].Y(),
-                                 scaleFactor.Z() * m_vertices[i].Z()));
+  for (auto vertex : m_vertices) {
```
Comment (OwenArnold) [inline Framework/Geometry/src/Objects/MeshObject.cpp]: I think should be `auto const &vertex` same with others.

Comment (matthew-d-jones) [reply Framework/Geometry/src/Objects/MeshObject.cpp]: :+1: I already pushed the change

File: Framework/Geometry/src/Objects/MeshObject.cpp [hunk 2]
```diff
@@ -400,12 +381,12 @@ double MeshObject::solidAngle(const Kernel::V3D &observer,
 {
   std::vector<V3D> scaledVertices;
   scaledVertices.reserve(m_vertices.size());
-  for (size_t i = 0; i < m_vertices.size(); ++i) {
-    scaledVertices.push_back(V3D(scaleFactor.X() * m_vertices[i].X(),
-                                 scaleFactor.Y() * m_vertices[i].Y(),
-                                 scaleFactor.Z() * m_vertices[i].Z()));
+  for (const auto &vertex : m_vertices) {
+    scaledVertices.emplace_back(scaleFactor.X() * vertex.X(),
```
Comment (LamarMoore) [inline Framework/Geometry/src/Objects/MeshObject.cpp]: 👍

**Review by LamarMoore — APPROVED** [review]
All looks fine to me.

**Pull request merged**